src/provider/form-factory-helper.ts

```diff
@@ -46,6 +46,11 @@ export function validateForm(context: FormContextProps) {
   return errors.length === 0;
 }
 
+export function validateEmptyForm(context: FormContextProps) {
+  const { methods: { formState: { dirtyFields } } } = context;
+  return Object.keys(dirtyFields).length === 0;
```

Comment:
Can you filter out [transient](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/e92c22c63f37c8b21cd5b97c39cef5b8d44dab31/src/types/schema.ts#L154) fields?